### PR TITLE
Small tweaks/additions to the stats endpoint

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

#707 

## Purpose/Implementation Notes

There were a few changes I wanted to make to the stats endpoint. These include:
* Fixing the total_size downloaded stat. It was broken because after processing data we delete it and set `is_downloaded` to False. This has been replaced by `input_data_size` and `output_data_size` which only include data that has actually been processed.
* Adds jobs count for successful jobs.
* Only considers successful jobs when calculating average job time.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

I tested this out locally, screenshot attached. Note that `output_data_size` is 0 because when running locally we don't push to S3 and that is what we're checking for.

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots

image

### CODE REVIEW

File: api/data_refinery_api/views.py [hunk 1]
```diff
@@ -1,66 +1,66 @@
...
+    SampleSerializer,
 
     # Job
-    SurveyJobSerializer,
     DownloaderJobSerializer,
     ProcessorJobSerializer,
+    SurveyJobSerializer,
 
     # Dataset
+    APITokenSerializer
```
Comment (Miserlou) [inline api/data_refinery_api/views.py]: This is creating a syntax error

**Comment by kurtwheeler** [comment]
@Miserlou I have made a ton of changes and this is now really ready for review. The new stats endpoint looks like:
```
{
    "survey_jobs": {
        "total": 23707,
        "pending": 23354,
        "completed": 347,
        "successful": 326,
        "open": 0,
        "average_time": 27.287414
    },
    "downloader_jobs": {
        "total": 6846,
        "pending": 6728,
        "completed": 96,
        "successful": 72,
        "open": 0,
        "average_time": 21.284127
    },
    "processor_jobs": {
        "total": 194,
        "pending": 78,
        "completed": 84,
        "successful": 60,
        "open": 6,
        "average_time": 59.568368
    },
    "samples": {
        "total": 6562
    },
    "experiments": {
        "total": 317
    },
    "input_data_size": 571874661,
    "output_data_size": 0,
    "active_volumes": [],
    "nomad_running_jobs": 0,
    "nomad_pending_jobs": 29,
    "nomad_running_jobs_by_type": {
        "SURVEYOR": 0,
        "NO_OP": 0,
        "DOWNLOADER": 0,
        "SMASHER": 0,
        "AFFY_TO_PCL": 0,
        "TRANSCRIPTOME_INDEX": 0,
        "ILLUMINA_TO_PCL": 0,
        "JANITOR": 0,
        "SALMON": 0
    },
    "nomad_pending_jobs_by_type": {
        "SURVEYOR": 0,
        "NO_OP": 0,
        "DOWNLOADER": 0,
        "SMASHER": 0,
        "AFFY_TO_PCL": 28,
        "TRANSCRIPTOME_INDEX": 0,
        "ILLUMINA_TO_PCL": 0,
        "JANITOR": 0,
        "SALMON": 1
    },
    "nomad_running_jobs_by_volume": {
        "volume_0": 0
    },
    "nomad_pending_jobs_by_volume": {
        "volume_0": 29
    }
}
```

A couple limitations that it has which I think I'm okay with:
  * Jobs that aren't specific to a volume will not be included in the `nomad_*_jobs_by_volume` metrics. (These include surveyor, downloader, and smasher)
  * I had to do some nomad job name munging to be able to quickly parse out the job types/index volumes. It might be a bit brittle when additional jobs types are added or if the job names ever change.

**Comment by Miserlou** [comment]
Tests are unhappy

File: infrastructure/nomad-configuration/client-instance-user-data.tpl.sh [hunk 2]
```diff
@@ -90,7 +90,7 @@ echo "
 # Cannot specify bip option in config file because it is hardcoded in
 # the startup command because docker is run by clowns.
 service docker stop
-nohup /usr/bin/dockerd -s overlay2 --bip=172.17.77.1/22 --log-driver=json-file --log-opt max-size=100m --log-opt max-file=3 > /dev/null &
+nohup /usr/bin/dockerd -s overlay2 --bip=172.17.77.1/22 --log-driver=json-file --log-opt max-size=100m --log-opt max-file=3 > /var/log/docker_daemon.log &
```
Comment (Miserlou) [inline infrastructure/nomad-configuration/client-instance-user-data.tpl.sh]: Unrotated logs will cause eventual instance death

**Pull request merged**